services/provider/server/server.go

```diff
@@ -590,16 +597,6 @@ func (s *OCSProviderServer) GetStorageClassClaimConfig(ctx context.Context, req
 			storageClassData["clusterID"] = s.namespace
 			storageClassData["radosnamespace"] = cephRes.Name
 
-			nodeCephClientSecret, _, err := s.getCephClientInformation(ctx, cephRes.CephClients["node"])
-			if err != nil {
-				return nil, status.Error(codes.Internal, err.Error())
-			}
-
-			provisionerCephClientSecret, _, err := s.getCephClientInformation(ctx, cephRes.CephClients["provisioner"])
-			if err != nil {
-				return nil, status.Error(codes.Internal, err.Error())
-			}
```

Comment:
just so I understand correctly, if ceph clients doesn't exist (or created) rather than failing here, we are pushing the handling to client side?

taking block as an example, suppose we fail to reconcile rbdprovisioner and rbdnode secrets for cephclient-node will not exist, is it fine?

only a question, not requesting a mandatory change.

Reply:
My point is a bit different, the clients are reconciled sequentially and in the refactor you are generating the secret names even w/o checking the Ceph client is created or not.

Switch might not help if the status for Ceph client isn't updated at all, then also you'll update storageclasses w/ not existing secret names.

The repercussion is choosing b/n these, if storageclass is created at client end we also have referenced secret created (existing) vs even if storageclass is created at client end we aren't guaranteed that referenced secret is also created (refractor).

Am I making sense?

Reply:
...not, I'm not following you. This function is just generating information to send to the consumer. So *prior* to anything being sent to the consumer we're already validating the existence of the CephClients, and this PR does not change that behavior. 

Before this PR, for a given StorageClassRequest, we loop through `Status.CephResources` and either 1. Validate the existence of a CephClient, inspect its existing provider-side Secret, and then forward that information for the consumer to create a Secret on its side with a new (generated) name. Or 2. For any rados namespace or subvolgroup, loop through its associated CephClients, inspect their Status for the existence of a Secret name.

However, at this point we no longer care about 2 since we're not telling the consumer to create Secrets based on the provider-side Secret names, and 1 takes care of ensuring the provider-side Secrets exist for both CephClients.

Am I still missing the point? Sorry if I am, it's been a rough morning.

Reply:
Just rechecked, the loop gets triggered only after phase is ready and so the scenario I mention will not be hit.

Only a minor comment, could you pls construct the name after the status check?

Reply:
Sure, done.